units/en/unit7/self-play.mdx

```diff
@@ -0,0 +1,137 @@
+
+- We **start with a copy of our agent as an opponent** this way this opponent is on a similar level.
+- We **learn from it**, and when we acquired some skills, we **update our opponent with a more recent copy of our training policy**.
+
+The theory behind self-play is not something new, it was already used by Arthur Samuel’s checker player system in the fifties, and by Gerald Tesauro’s TD-Gammon in 1955. If you want to learn more about the history of self-play [check this very good blogpost by Andrew Cohen](https://blog.unity.com/technology/training-intelligent-adversaries-using-self-play-with-ml-agents)
+
+## Self-Play in MLAgents
+
+Self-Play is integrated into the MLAgents library and is managed by multiple hyperparameters that we’re going to study. But the main focus as explained in the documentation is the **tradeoff between the skill level and generality of the final policy and the stability of learning**.
+
+Training against a set of slowly changing or unchanging adversaries with low diversity **results in more stable training. But a risk to overfit if the change is too slow.**
+
+We need then to control:
+
+- How **often do we change opponents** with `swap_steps` and `team_change` parameters.
+- The **number of opponents saved** with `window` parameter. A larger value of `window`
+ means that an agent's pool of opponents will contain a larger diversity of behaviors since it will contain policies from earlier in the training run.
+- **Probability of playing against the current self vs opponent** sampled in the pool with `play_against_latest_model_ratio`. A larger value of `play_against_latest_model_ratio`
+ indicates that an agent will be playing against the current opponent more often.
+- The **number of training steps before saving a new opponent** with `save_steps` parameters. A larger value of `save_steps`
+ will yield a set of opponents that cover a wider range of skill levels and possibly play styles since the policy receives more training.
+
+To get more details about these hyperparameters you definitely need [to check this part of the documentation](https://github.com/Unity-Technologies/ml-agents/blob/develop/docs/Training-Configuration-File.md#self-play)
+
+
+## The ELO Score to evaluate our agent
+
+### What is ELO Score?
+
+In adversarial games, tracking the **cumulative reward is not always a meaningful metric to track the learning progress:** because this metric is **dependent only on the skill of the opponent.**
+
+Instead, we’re using an ***ELO rating system*** (named after Arpad Elo) that calculates the **relative skill level** between 2 players from a given population in a zero-sum game.
+
+In a zero-sum game: one agent wins is the other agent loss. It’s a mathematical representation of a situation in which each participant’s gain or loss of utility **is exactly balanced by the gain or loss of the utility of the other participants.** We talk about zero-sum games because the sum of utility is equal to zero.
```

Comment:
```suggestion
In a zero-sum game: one agent wins, and the other agent loses. It’s a mathematical representation of a situation in which each participant’s gain or loss of utility **is exactly balanced by the gain or loss of the utility of the other participants.** We talk about zero-sum games because the sum of utility is equal to zero.
```